contracts/FundManagement.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * Contract to allow Tracer DAO to delegate funds to managers to spend as they see fit. 
+ * Managers can request to withdraw funds assigned to them. The DAO may take back the funds at any time.
+ */
+contract FundManagement is Ownable {
+    struct Fund {
+        uint256 totalAmount; // Total amount of tokens assigned to the manager
+        address asset;
+        uint256 requestedWithdrawTime; // Timestamp which the manager may withdraw the pendingWithdrawAmount
+        uint256 pendingWithdrawAmount; // Total amount of tokens the manager is pending to withdraw
+    }
+
+    using SafeERC20 for IERC20;
+
+    uint256 public requestWindow = 2 days;
+    mapping(address => mapping(uint256 => Fund)) public funds; // user -> fundId -> fund
+    mapping(address => uint256) public numberOfFunds; // user -> number of funds owned
+    mapping(address => uint256) public locked; // asset -> amount locked up
+
+    constructor() {}
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+
+    /**
+     * @notice User requests funds that are allocated to him. After the request window if no clawback, they may claim those requested  funds.
+     */
+    function requestFunds(uint256 fundNumber, uint256 amount) external {
+        require(
+            fundNumber >= 0 && fundNumber < numberOfFunds[msg.sender],
+            "The input fund number does not exist"
+        );
+        Fund storage fund = funds[msg.sender][fundNumber];
+        uint256 totalWithdrawableAmount = fund.pendingWithdrawAmount + amount;
+        require(
+            totalWithdrawableAmount <= fund.totalAmount, 
+            "Requested amount plus withdrawable amount greater than total allocated funds"
+        );
+        
+        fund.requestedWithdrawTime = block.timestamp + requestWindow;
+        fund.pendingWithdrawAmount = totalWithdrawableAmount;
+
+        emit RequestFunds(msg.sender, amount);
+    }
+
+    /**
+     * @notice User claims the funds they requested. Only claimable after request window has passed with no clawbacks.
+     */
+    function claim(uint256 fundNumber) external {
+        require(
+            fundNumber >= 0 && fundNumber < numberOfFunds[msg.sender],
+            "The input fund number does not exist"
+        );
+        Fund storage fund = funds[msg.sender][fundNumber];
```

Comment:
I believe you can save gas by making this `fund` a `memory` variable, and then changing lines `76` and `77` to 
```
funds[msg.sender][fundNumber].totalAmount = fund.totalAmount - pendingAmount;
funds[msg.sender][fundNumber].pendingWithdrawAmount = 0;
```
But run a quick test locally with both my suggestion and the current implementation to see if it does actually reduce gas cost.

This might also be true for `requestFunds`, but I'm not convinced since it only reads from the variable twice.

Reply:
used the hardhat gas reporter on my unit tests seems like this make it use more gas

Reply:
Interesting. Although hardhat gas reporter uses USD values and is therefore inaccurate. Better to calculate the total amount of gas used rather than USD cost.

Either way I wasn't too sure so I'm easy.